src/transformations/transform-constructor.ts

```diff
@@ -82,11 +94,18 @@ export function* transformConstructor(
       argNames: string[] = [],
     ) => [
       `function __${name}_init(${argsList}) internal onlyInitializing {`,
-      buildSuperCallsForChain2(contractNode, tools, helper),
+      buildSuperCallsForChain(contractNode, tools, helper),
       emptyConstructor ? [] : [`__${name}_init_unchained(${argNames.join(', ')});`],
       `}`,
       ``,
-      `function __${name}_init_unchained(${unchainedArgsList}) internal onlyInitializing {`,
+      [
+        `function`,
+        ` __${name}_init_unchained(${unchainedArgsList}) `,
+        `internal onlyInitializing`,
+        `${
+          modifiers && modifiers.length ? modifiers.map(m => ' ' + helper.read(m)).join(' ') : ''
+        } {`,
+      ].join(''),
```

Comment:
The suggestion was to join with spaces, it makes the code much simpler.

`modifiers` should have a default value in the function definition.

```suggestion
      [
        `function`,
        `__${name}_init_unchained(${unchainedArgsList})`,
        `internal onlyInitializing`,
        ...modifiers.map(m => helper.read(m)),
        `{`,
      ].join(' '),
```